From 738a0d76c3102509e052927e49f978b197bc3a3c Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Fri, 6 Dec 2019 19:19:30 +0000
Subject: [PATCH 4/9] ozone/wayland: Fix regression with hiding nested windows.

It must be non-tooltip that closes children windows.

Bug: 1028046
Change-Id: I9b5212873cb2c1a4179a28e915edd5fe84b29c5e
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1954483
Reviewed-by: Michael Spang <spang@chromium.org>
Commit-Queue: Maksim Sisov <msisov@igalia.com>
Cr-Commit-Position: refs/heads/master@{#722560}
---
 ui/ozone/platform/wayland/host/wayland_window.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/ozone/platform/wayland/host/wayland_window.cc b/ui/ozone/platform/wayland/host/wayland_window.cc
index 709fb681f7d9..1f8cc423d4aa 100644
--- a/ui/ozone/platform/wayland/host/wayland_window.cc
+++ b/ui/ozone/platform/wayland/host/wayland_window.cc
@@ -247,7 +247,7 @@ void WaylandWindow::Show(bool inactive) {
 }
 
 void WaylandWindow::Hide() {
-  if (!is_tooltip_) {
+  if (is_tooltip_) {
     tooltip_subsurface_.reset();
   } else {
     if (child_window_)
-- 
2.24.1

